lib/logstash/outputs/graphite.rb

```diff
@@ -18,17 +18,24 @@ class LogStash::Outputs::Graphite < LogStash::Outputs::Base
   METRIC_PLACEHOLDER = "*"
 
   # The hostname or IP address of the Graphite server.
-  config :host, :validate => :string, :default => "localhost"
+  config :host, :validate => :string, :default => "localhost", :deprecated => "This setting is being deprecated, use :hosts instead."
 
-  # The port to connect to on the Graphite server.
-  config :port, :validate => :number, :default => 2003
+  # The port to connect to the Graphite server.
+  config :port, :validate => :number, :default => 2003, :deprecated => "This setting is being deprecated, use :hosts instead."
 
-  # Interval between reconnect attempts to Carbon.
+  # The list of known Graphite servers to connect to. One host is selected randomly (there is no precedence). If one host becomes unreachable, another one is selected randomly.
+  # All entries in this list can contain a port number. If no port number is given, the default value of 2003 is used.
+  config :hosts, :validate => :string, :list => true, :default => ["localhost:2003"]
+
+  # Interval between reconnect attempts to Graphite server, seconds.
   config :reconnect_interval, :validate => :number, :default => 2
 
   # Should metrics be resent on failure?
   config :resend_on_failure, :validate => :boolean, :default => false
 
+  # Number of attempts to be made resending metrics before abandoning
+  config :resend_attempts, :validate => :number, :default => 3
```

Comment:
This resend-attempts feature seems out of scope for the proposed change (add multiple hosts, choose at random). Can you move it to a separate PR?